### PR TITLE
Angular: Update angular to v12

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

**Builds on #18034**

Tried to add Angular12 to monorepo by adapting https://github.com/storybookjs/storybook/pull/15184 to our latest tooling changes in  #18034 (which updates to latest TS/webpack5/etc.). Work in progress.

## How to test

- [ ] CI passes

### CODE REVIEW

**Comment by kroeder** [comment]
@ndelangen @shilman I got it working with Angular 13 as discussed with you @shilman 

However, please review the two commits that contain "EXPERIMENTAL". I wrote a resume about what's going on in each commit and I have no clue how to fix it right now. I hope this helps! 

You can fix them independently by just reverting one of them - all 3 commits are atomic.

File: app/angular/src/server/angular-devkit-build-webpack.ts [hunk 1]
```diff
@@ -145,6 +145,7 @@ const buildWebpackConfigOptions = async (
   };
 
   return {
+    projectName: 'this-is-just-a-fake-name-for-getting-rid-of-the-error',
```
Comment (kroeder) [inline app/angular/src/server/angular-devkit-build-webpack.ts:148]: Probably should change that

**Comment by ndelangen** [comment]
@kroeder @shilman I rebased it and merged in my modern-frameworks branch, let's see what happens.

**Comment by kroeder** [comment]
> @kroeder @shilman I rebased it and merged in my modern-frameworks branch, let's see what happens.

Don't forget to change [4 days ago](https://github.com/storybookjs/storybook/pull/18035#pullrequestreview-966845476)

**Pull request merged**